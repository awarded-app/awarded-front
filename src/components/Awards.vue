<template>
  <div>
    <spinner v-if="$apollo.loading"/>
    <div v-else-if="awards">
      <award v-for="award in awards.nodes" :key="award.id" :award="award"/>
    </div>
    <div v-else>Error...</div>
  </div>
</template>

<script>
import gql from "graphql-tag";
import Spinner from "./Spinner";
import Award from "./Award";
export default {
  name: "Awards",
  components: { Award, Spinner },
  data() {
    return {
      awards: null
    };
  },
  apollo: {
    awards: {
      query: gql`
        {
          awards {
            totalCount
            nodes {
              ...award
            }
          }
        }
        ${Award.fragments.award}
      `
    }
  }
};
</script>

<style lang="scss" scoped></style>
