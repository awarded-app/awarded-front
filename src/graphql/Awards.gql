#import "./AwardFragment.gql"

query awards {
  awards {
    totalCount
    nodes {
      ...award
    }
  }
}
