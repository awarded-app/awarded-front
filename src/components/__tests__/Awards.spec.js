import { shallowMount } from "@vue/test-utils";
import Awards from "@/components/Awards.vue";

test("sanity test", () => {
  return;
});
