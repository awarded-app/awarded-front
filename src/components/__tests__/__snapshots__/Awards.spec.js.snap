// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Awards.vue displayed awards correctly with query data 1`] = `
<div>
  <div>
    <award-stub
      award="[object Object]"
    />
    <award-stub
      award="[object Object]"
    />
    <award-stub
      award="[object Object]"
    />
  </div>
</div>
`;
